modules/gltf/src/lib/extensions/EXT_structural_metadata.ts

```diff
@@ -701,3 +702,220 @@ function getEnumByValue(
 
   return null;
 }
+
+/*
+  Encoding data
+*/
+
+export interface TypedFeatureAttribute {
+  name: string;
+  valueType: string;
+  valuesPerElement: number;
+  values: number[] | string[];
+}
+
+const SCHEMA_CLASS_ID_DEFAULT = 'schemaClassId';
+
+/**
+ * Creates ExtStructuralMetadata, creates the schema and creates a property table containing feature data provided.
+ * @param scenegraph - Instance of the class for structured access to GLTF data.
+ * @param typedFeatureAttributes - Feature attributes
+ * @param classId - classId to use for encoding metadata.
+ * @returns Index of the table created.
+ */
+export function createExtStructuralMetadata(
+  scenegraph: GLTFScenegraph,
+  typedFeatureAttributes: TypedFeatureAttribute[],
```

Comment:
The extension should work with `encode` function (similar to `EXT_MESH_FEATURES`).
We need th `create...` function but locate it after `encode`.
featureAttributes is an I3S entities. We need something more generic. Look at the `propertyTable` which is an output of this extension `decode`.

Reply:
It will be addressed in a separate PR for the EXT_structural _metadata.